aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1601,6 +1601,154 @@ Tensor& linalg_norm_out(Tensor& result, const Tensor& self, std::string ord, opt
+  // we can't get the result for all other (possibly) invertible matrices in the batch without an explicit for loop.
+  // This should change when at::inverse works with silent errors
+  if (self.dim() > 2) {
+    TORCH_CHECK(false,
+      "At least one matrix in the batch is not invertible, its condition number is infinity, "
+      "linalg_cond does not support yet calculating the condition number for all other (possibly invertible) matrices in the batch.");
+  }
+  auto result_shape = self.sizes().vec();
+  result_shape.pop_back();
+  result_shape.pop_back();  // result's shape is equal to self.shape[0:-2]
+  Tensor result = at::empty(result_shape, self.options());
+  at::fill_(result, INFINITY);
+  return result;
+}
+
+// Numerical or None norms
+Tensor linalg_cond(const Tensor& self, optional<Scalar> opt_ord) {
+  TORCH_CHECK(self.numel() > 0, "linalg_cond is not defined for empty tensors.");
+  TORCH_CHECK(self.dim() >= 2, "linalg_cond only supports matrices or batches of matrices, but got a tensor with ",
+    self.dim(), " dimensions.");
+
+  // The default case is using 2-norm
+  Scalar ord = opt_ord.has_value() ? opt_ord.value() : 2;
+
+  // If ord == None or ord == ±2
+  if (std::abs(ord.toDouble()) == 2.0) {
+    auto singular_values = std::get<1>(at::svd(self));
+    auto s_max = std::get<0>(singular_values.max(/*dim=*/-1));
+    auto s_min = std::get<0>(singular_values.min(/*dim=*/-1));
+    Tensor result;
+    if (ord.toDouble() == -2.0) {
+      result = s_min / s_max;
+    } else {
+      result = s_max / s_min;
+    }
+    // convert FLT_MAX or DBL_MAX to INFINITY for NumPy compatibility
+    switch (result.scalar_type()) {
+      case ScalarType::Double: {
+        Scalar dbl_max = std::numeric_limits<double>::max();  // DBL_MAX
+        Scalar inf = std::numeric_limits<double>::infinity();  // HUGE_VAL
+        return at::where(result == dbl_max, inf, result);
+      }
+      case ScalarType::Float: {
+        Scalar flt_max = std::numeric_limits<float>::max();  // FLT_MAX
+        float inf = std::numeric_limits<float>::infinity();  // HUGE_VALF
+        // Scalar.dtype() is always ScalarType::Double for isFloatingPoint() = true
+        // and at::where doesn't allow arguments with different dtype
+        // so let's use 0-dim tensor filled with inf
+        Tensor inf_tensor = at::empty({}, result.options());
+        at::fill_(inf_tensor, inf);
+        return at::where(result == flt_max, inf_tensor, result);
+      }
+      default:
+        TORCH_CHECK(false, "linalg_cond got an unexpected result type ", toString(result.scalar_type()));
+    }
+  }
+
+  // ord == ±1 ord == ±inf
+  // since at::inverse is used in the implementation, self has to be a tensor consisting of square matrices
+  // the same check as squareCheckInputs(self) but with a slightly more informative error message
+  TORCH_CHECK(self.size(-1) == self.size(-2),
+              "linalg_cond only supports square matrices or batches of square matrices "
```

Comment:
See comment below, too. It's only some norms that only supports square matrices.